lib/ansible/modules/network/cnos/cnos_l3_interface.py

```diff
@@ -0,0 +1,425 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+#
+# Copyright (C) 2017 Lenovo, Inc.
+# (c) 2017, Ansible by Red Hat, inc
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Module to work on Link Aggregation with Lenovo Switches
+# Lenovo Networking
+#
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: cnos_l3_interface
+version_added: "2.8"
+author: "Anil Kumar Muraleedharan (@amuraleedhar)"
+short_description: Manage Layer-3 interfaces on Lenovo CNOS network devices.
+description:
+  - This module provides declarative management of Layer-3 interfaces
+    on CNOS network devices.
+notes:
+  - Tested against CNOS 10.8.1
+options:
+  name:
+    description:
+      - Name of the Layer-3 interface to be configured eg. GigabitEthernet0/2
+  ipv4:
+    description:
+      - IPv4 address to be set for the Layer-3 interface mentioned in I(name)
+        option. The address format is <ipv4 address>/<mask>, the mask is number
+        in range 0-32 eg. 10.241.107.1/24
+  ipv6:
+    description:
+      - IPv6 address to be set for the Layer-3 interface mentioned in I(name)
+        option. The address format is <ipv6 address>/<mask>, the mask is number
+        in range 0-128 eg. fd5d:12c9:2201:1::1/64
+  aggregate:
+    description:
+      - List of Layer-3 interfaces definitions. Each of the entry in aggregate
+        list should define name of interface C(name) and a optional C(ipv4) or
+        C(ipv6) address.
+  state:
+    description:
+      - State of the Layer-3 interface configuration. It indicates if the
+        configuration should be present or absent on remote device.
+    default: present
+    choices: ['present', 'absent']
+  provider:
+    description:
+      - B(Deprecated)
+      - "Starting with Ansible 2.5 we recommend using
+         C(connection: network_cli)."
+      - For more information please see the
+        L(CNOS Platform Options guide, ../network/user_guide/platform_cnos.html).
+      - HORIZONTALLINE
+      - A dict object containing connection details.
+    version_added: "2.8"
```

Comment:
this is redundant, the whole module was added in 2.8

Reply:
Done